src/main/java/org/codehaus/plexus/components/io/filemappers/SuffixFileMapper.java

```diff
@@ -0,0 +1,81 @@
+package org.codehaus.plexus.components.io.filemappers;
+
+/*
+ * Copyright 2007 The Codehaus Foundation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import javax.annotation.Nonnull;
+
+import org.codehaus.plexus.util.StringUtils;
+
+/**
+ * A file mapper, which maps by adding a suffix.
```

Comment:
Would be nice if make it clear that the suffix is added before the dot. It is mentioned in the site docs but it would be nice if you state it here as well (so it is visible in the IDEs).